### PR TITLE
Infrastructure/#4666 - Fix GTM stories integration with Analytics

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #4666 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: stories/modules-tagmanager-setup.stories.js [hunk 1]
```diff
@@ -49,7 +49,51 @@ import {
...
+
+	const { buildAndReceiveWebAndAMP } = createBuildAndReceivers( registry );
+	buildAndReceiveWebAndAMP( {
+		accountID,
+		webPropertyID: propertyID,
+		ampPropertyID: propertyID,
+	} );
+}
+
+function provideWebContainer( registry, accountID ) {
```
Comment (aaemnnosttv) [inline stories/modules-tagmanager-setup.stories.js:70]: I'd expect a `provide` helper to receive the container itself as well. This is essentially just selecting the first web container that is available for the given account. This is fine as a utility, I'd just name it differently to make that a bit more clear. Perhaps something like `selectFirst(Web|AMP)Container`?

File: stories/modules-tagmanager-setup.stories.js [hunk 2]
```diff
@@ -305,6 +349,13 @@ storiesOf( 'Tag Manager Module/Setup', module )
 							accountID,
 							internalContainerID,
 						} );
+
+					buildAndReceiveProperties(
+						registry,
+						accountID,
+						internalContainerID
+					);
+					provideWebContainer( registry, accountID );
```
Comment (aaemnnosttv) [inline stories/modules-tagmanager-setup.stories.js:358]: This call doesn't use any data from the parsed container version so it should go below (outside of this callback), same with other related calls.

File: stories/modules-tagmanager-setup.stories.js [hunk 3]
```diff
@@ -49,7 +49,51 @@ import {
 import { MODULES_ANALYTICS } from '../assets/js/modules/analytics/datastore/constants';
 import * as fixtures from '../assets/js/modules/tagmanager/datastore/__fixtures__';
 import * as modulesFixtures from '../assets/js/googlesitekit/modules/datastore/__fixtures__';
-import { parseLiveContainerVersionIDs } from '../assets/js/modules/tagmanager/datastore/__factories__/utils';
+import {
+	createBuildAndReceivers,
+	parseLiveContainerVersionIDs,
+} from '../assets/js/modules/tagmanager/datastore/__factories__/utils';
+
+function buildAndReceiveProperties( registry, accountID, internalContainerID ) {
```
Comment (aaemnnosttv) [inline stories/modules-tagmanager-setup.stories.js:57]: This is a bit confusing as named as it's not receiving properties (GA/Search Console entities) but building and receiving GTM containers with properties configured.

It's also a bit odd that it seems to depend on the existing live container version.

Let's inline this utility as it's a bit hard to understand by itself and the part below is already quite minimal.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @hussain-t, this is nearly ready just a few points to address.

File: stories/modules-tagmanager-setup.stories.js (hunk 3 above)
Comment (hussain-t) [reply stories/modules-tagmanager-setup.stories.js]: Updated ✔️

File: stories/modules-tagmanager-setup.stories.js (hunk 2 above)
Comment (hussain-t) [reply stories/modules-tagmanager-setup.stories.js]: Good catch. Updated ✔️

File: stories/modules-tagmanager-setup.stories.js (hunk 1 above)
Comment (hussain-t) [reply stories/modules-tagmanager-setup.stories.js]: Updated ✔️

**Pull request merged**